lib/archethic_web/controllers/api/web_hosting_controller.ex

```diff
@@ -109,6 +111,25 @@ defmodule ArchethicWeb.API.WebHostingController do
     end)
   end
 
+  defp get_single_file_name(json_content) do
+    keys = Map.keys(json_content)
+
+    case Enum.count(keys) do
+      1 ->
+        # Control if it is a file or a folder
+        file_name = Enum.at(keys, 0)
+
+        if Map.get(json_content, file_name) |> Map.has_key?("content") do
```

Comment:
I think you need here to specify a default map if not found otherwise `Map.has_key?` will fail if it's `nil`

Reply:
Does `Map.get` can fail ? As `file_name` is the first key of the map

Reply:
It doesn't fail but return nil if not found

Reply:
Unless you pass a third parameter being the default value.

Reply:
Yes indeed it should not